src/js/controller.js

```diff
@@ -0,0 +1,771 @@
+import { window, document } from './utils/server-safe-globals.js';
+import { fullscreenApi } from './utils/fullscreenApi.js';
+import { containsComposedNode } from './utils/element-utils.js';
+import { serializeTimeRanges } from './utils/time.js';
+import {
+  hasVolumeSupportAsync,
+  fullscreenSupported,
+  pipSupported,
+  airplaySupported,
+  castSupported
+} from './utils/platform-tests.js';
+import {
+  MediaUIAttributes,
+  TextTrackKinds,
+  TextTrackModes,
+  AvailabilityStates,
+  StreamTypes,
+} from './constants.js';
+
+import {
+  stringifyTextTrackList,
+  getTextTracksList,
+  updateTracksModeTo,
+} from './utils/captions.js';
+
+let volumeSupported;
+export const volumeSupportPromise = hasVolumeSupportAsync().then((supported) => {
```

Comment:
NOTE: Currently exporting this based on our status quo impl of checking/propagating volume support in MediaController.